JavaScript (eslint)/eslint-config-hzdg-react/rules/react.js

```diff
@@ -128,7 +131,7 @@ module.exports = {
     'react/prefer-es6-class': ['error', 'always'],
     'react/prefer-stateless-function': ['error', {ignorePureComponents: true}],
     'react/react-in-jsx-scope': 'error',
-    'react/require-default-props': 'error',
+    'react/require-default-props': 'warn',
```

Comment:
I'd rather disable a rule than have it warn. IMO, in any case where a warning might be shown, it's better to make it an error and force the developer to fix or override the error. Warnings make it ok to ignore, and once we're ignoring warnings, we're bound to start ignoring errors, too.

It's easy enough to just disable a rule by line or file (or even project), so in general, i don't think warnings are useful.

That said, in this case, i'm leaning slightly toward disabling this rule. It's useful to have if you are using propTypes, but i can't think of an easy way to automate disabling it if you happen to be using flowtype or typescript.

Since i'm (perhaps optimistically) anticipating more reliance on typed JS in our projects, i think we could disable this now, and look to adding flow or typescript to new projects by default.

Then, if we accept the premise that we're looking at typed JS to solve component type checking, maybe we disable all the proptype-related rules?

Thoughts?

Reply:
I've disabled the rule(for now) and will open an issue to discuss this across the team which we can update further on